pkg/webhook/admission/webhook.go

```diff
@@ -138,11 +143,32 @@ func (wh *Webhook) InjectLogger(l logr.Logger) error {
 	return nil
 }
 
+// WithRecoverPanic takes a bool flag which indicates whether the panic caused by webhook should be recovered.
+func (wh *Webhook) WithRecoverPanic(recoverPanic bool) *Webhook {
+	wh.RecoverPanic = recoverPanic
+	return wh
+}
+
 // Handle processes AdmissionRequest.
 // If the webhook is mutating type, it delegates the AdmissionRequest to each handler and merge the patches.
 // If the webhook is validating type, it delegates the AdmissionRequest to each handler and
 // deny the request if anyone denies.
-func (wh *Webhook) Handle(ctx context.Context, req Request) Response {
+func (wh *Webhook) Handle(ctx context.Context, req Request) (response Response) {
+	defer func() {
+		if r := recover(); r != nil {
+			if wh.RecoverPanic {
```

Comment:
Shouldn't this `if` condition be before the `recover()`?

Reply:
We won't be able to process the panic if we don't recover from it first. Panic recovery works the same way in controller.go: https://github.com/kubernetes-sigs/controller-runtime/blob/15154aaa67679df320008ed45534f83ff3d6922d/pkg/internal/controller/controller.go#L108

Reply:
But we don't want to process it when RecoverPanic is false? What is done in controller.go seems quite pointless to me and IIRC the `r` won't contain the stacktrace?

Reply:
Makes sense, let's make `RecoverPanic` flag more fair and really not process anything if it is set to false. Adjusted my PR accordingly 👍